### PR TITLE
Split pilot node metadata from bootstrap only meta

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/24024

Currently we have a big struct of metadata that keeps growing. Some of
this is used for Pilot, while others is used for envoy filters. In order
to avoid scope creep for pilot's dependencies on node metadata, this
splits the structs so that Pilot has an explicitly defined list of
metadata it uses for config gen.

Note: Pilot can trivially add metadata to its struct in a backwards
compatible way. This does *not* change the json format, only internal
structures. This is intended only to make it an explicit decision when
we add a dependency, rather than accidentally depending on something new
just because it is present already

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
@istio/wg-networking-maintainers

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -371,6 +371,36 @@ func (s *NodeMetaProxyConfig) UnmarshalJSON(data []byte) error {
 	return gogojsonpb.Unmarshal(bytes.NewReader(data), pc)
 }
 
+// BootstrapNodeMetadata is a superset of NodeMetadata, intended to model the entirety of the node metadata
+// we configure in the Envoy bootstrap. This is split out from NodeMetadata to explicitly segment the parameters
+// that are consumed by Pilot from the parameters used only as part of the bootstrap. Fields used by bootstrap only
+// are consumed by Envoy itself, such as the telemetry filters.
+type BootstrapNodeMetadata struct {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/context.go:378]: node seems kind of duplicate, how about calling `BootstrapMetadata`

**Pull request merged**